src/test/kotlin/me/spradling/gift/core/tests/unit/api/GetAccountsHandlerTests.kt

```diff
@@ -0,0 +1,99 @@
+package me.spradling.gift.core.tests.unit.api
+
+import io.netty.handler.codec.http.HttpResponseStatus
+import me.spradling.gift.core.api.extensions.unwrap
+import me.spradling.gift.core.api.models.Account
+import me.spradling.gift.core.api.models.ApiRequest
+import me.spradling.gift.core.api.models.errors.ErrorDetails
+import me.spradling.gift.core.api.routes.v1.GetAccountsHandler
+import me.spradling.gift.core.tests.unit.UnitTestBase
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.*
+import org.mockito.Mockito.`when`
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DisplayName("When I call `handleRequest` on GetAccountsHandler,")
+internal class GetAccountsHandlerTests : UnitTestBase() {
+
+  private val getHandler = GetAccountsHandler(inMemoryStorageClient)
+
+  @Nested
+  @DisplayName("given no results")
+  inner class NoResults {
+
+    @Test
+    @DisplayName("request should fail with Not Found error")
+    fun failsWithNotFoundError() {
+      val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap()
+
+      verifyErrorResponse(response, ErrorDetails.RESOURCE_NOT_FOUND)
+    }
+  }
+
+  @Nested
+  @DisplayName("given a valid request")
+  inner class ValidRequest {
+
+    @BeforeEach
+    fun setup() {
+      inMemoryStorageClient.createAccount(storageConverter.convert(validAccounts["api"]!!))
+    }
+
+    @Nested
+    @DisplayName("given valid query parameters,")
+    inner class ValidQueryParameter {
+
+      @BeforeEach
+      fun setup() {
+        `when`(mockContext.queryParam("limit")).thenReturn(listOf("10")) // Return that we want 10 accounts when calling the "limit" query param
+      }
+
+      @Test
+      @DisplayName("should succeed with 200 code")
+      fun shouldReturnAccounts() {
+        val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap() // Execute request
+
+        assertThat(response.statusCode).isEqualTo(HttpResponseStatus.OK.code()) // Should have 200 code
+        assertThat(response.body.isPresent).isEqualTo(true)
+        val retrievedAccounts = response.body.get() as List<Account>
+        assertThat(listOf(validAccounts["api"]).containsAll(retrievedAccounts) &&
```

Comment:
```Java lists implement equals method and two lists are defined to be equal if they contain the same elements in the same order. I guess, you are missing equals method in your MyObject class.```

You should really just be able to do, `assertThat(listOf(validAccounts["api"]).isEqualTo(response.body.get())`